### PR TITLE
feat(volume) Add the update snapshot data integrity option

### PR DESCRIPTION
**Pull request by smallteeths**
Signed-off-by: Siye Wang <siye@rancher.com>

https://github.com/longhorn/longhorn/issues/4994

### CODE REVIEW

File: src/routes/volume/VolumeBulkActions.js [hunk 1]
```diff
@@ -160,6 +163,7 @@ function bulkActions({ selectedRows, engineImages, bulkDeleteVolume, showBulkEng
     { key: 'expandVolume', name: 'Expand Volume', disabled() { return selectedRows.length === 0 || selectedRows.some((item) => item.state !== 'detached') || !conditionsScheduled() } },
     { key: 'updateBulkReplicaCount', name: 'Update Replicas Count', disabled() { return selectedRows.length === 0 || isHasStandy() || disableUpdateBulkReplicaCount() || upgradingEngine() } },
     { key: 'updateBulkDataLocality', name: 'Update Data Locality', disabled() { return selectedRows.length === 0 || isHasStandy() || disableUpdateBulkDataLocality() || upgradingEngine() } },
+    { key: 'updateSnapshotDataIntegrity', name: 'Update Snapshot Data Integrity', disabled() { return selectedRows.length === 0 || selectedRows.some((item) => item.state !== 'attached') } },
```
Comment (innobead) [inline src/routes/volume/VolumeBulkActions.js:166]: Why not `selectedRows.some((item) => attachable(item))`? I assume all volumes should be detached when updating this bulk setting?

**Review by innobead — COMMENTED** [review]
In general, LGTM.

File: src/routes/volume/detail/index.js [hunk 2]
```diff
@@ -40,8 +41,13 @@ function VolumeDetail({ snapshotModal, dispatch, backup, engineimage, eventlog,
   const currentBackingImage = selectedVolume && selectedVolume.backingImage && backingImage.data ? backingImage.data.find(item => item.name === selectedVolume.backingImage) : null
   const settings = setting.data
   const defaultDataLocalitySetting = settings.find(s => s.id === 'default-data-locality')
+  const defaultsnapshotDataIntegritySetting = settings.find(s => s.id === 'snapshot-data-integrity')
```
[the remark this answers is not in the document]
Comment (smallteeths) [reply src/routes/volume/detail/index.js]: done

File: src/routes/volume/index.js [hunk 3]
```diff
@@ -119,6 +122,11 @@ class Volume extends React.Component {
     const defaultDataLocalityOption = defaultDataLocalitySetting && defaultDataLocalitySetting.definition && defaultDataLocalitySetting.definition.options ? defaultDataLocalitySetting.definition.options : []
     const defaultDataLocalityValue = defaultDataLocalitySetting && defaultDataLocalitySetting.value ? defaultDataLocalitySetting.value : 'disabled'
     const defaultRevisionCounterValue = defaultRevisionCounterSetting && defaultRevisionCounterSetting.value && defaultRevisionCounterSetting.value === 'true'
+    const defaultSnapshotDataIntegrityOption = defaultsnapshotDataIntegritySetting?.definition?.options ? defaultsnapshotDataIntegritySetting.definition.options.map((item) => { return { key: item.firstUpperCase(), value: item } }) : []
+    if (defaultSnapshotDataIntegrityOption.length > 0) {
+      defaultSnapshotDataIntegrityOption.push({ key: 'Ignored (Follow the global setting)', value: 'ignored' })
+    }
+    const defaultSnapshotDataIntegrityValue = defaultsnapshotDataIntegritySetting?.value ? defaultsnapshotDataIntegritySetting?.value : 'disabled'
```
[the remark this answers is not in the document]
Comment (smallteeths) [reply src/routes/volume/index.js]: done

File: src/routes/volume/VolumeBulkActions.js (hunk 1 above)
Comment (smallteeths) [reply src/routes/volume/VolumeBulkActions.js]: done

**Review by innobead — APPROVED** [review]

**Pull request merged**